.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,164 @@
+
+Generally speaking, the goal of Requests is to provide an interface that smoothes the differences across various server setups.
+That means there shouldn't be any code relying on any optional PHP extensions (even if they're common) unless we also provide a pure-PHP version (that is, we can use extensions to improve performance when available).
+Only two extensions are currently used:
+* [cURL][curl] is only used for better performance over sockets.
+* [JSON][json] for decoding received responses.
+
+[wp-requirements]: https://wordpress.org/about/requirements/
+[semver]: https://semver.org/
+[curl]: https://www.php.net/book.curl
+[json]: https://www.php.net/book.json
+
+## Linting
+
+Code should (obviously) not contain any parse errors.
+This project uses [PHP Parallel Lint][] to monitor this for all supported PHP versions.
+
+The linter can be run locally using `composer lint`.
+
+[PHP Parallel Lint]: https://github.com/php-parallel-lint/PHP-Parallel-Lint
+
+## Coding Style
+
+Please follow the existing coding style and best practices.
+This project uses [PHP_CodeSniffer][] to detect coding standard violations and apply automated fixes (whenever possible).
+
+* All files can be checked for coding standard violations by running `composer checkcs`.
+* Any automatically applicable fixes can be applied by running `composer fixcs`.
+
+[PHP_CodeSniffer]: https://github.com/squizlabs/PHP_CodeSniffer
+
+## Unit Tests
+
+PRs should include unit tests for all changes.
+
+Tests for this library are written using [PHPUnit][] in combination with the [PHPUnit Polyfills][].
+This means that tests can be written for the latest version of PHPUnit (9.x at the time of writing) and still be run on all PHPUnit versions needed to test all supported PHP versions (PHPUnit 5.x - 9.x).
+
+Tests are organized as follows:
+* Each class in the `src` directory has a corresponding subdirectory in the `tests` directory.
+    Example: the tests for the `\WpOrg\Requests\Cookie` class in `src/Cookie.php` can be found in the `tests/Cookie/` subdirectory.
+* Unit tests for a specific method will be in this subdirectory in a file/class matching the method name.
+    Example: unit tests for the `\WpOrg\Requests\Cookie::domain_matches()` method are in the `WpOrg\Requests\Tests\Cookie\DomainMatchesTest` class in the `tests/Cookie/DomainMatchesTest.php` file.
+* Integration tests for a class will be in this subdirectory in a file/class matching the class name.
+    Example: integration tests for the `\WpOrg\Requests\Cookie` class are in the `WpOrg\Requests\Tests\Cookie\CookieTest` class in the `tests/Cookie/CookieTest.php` file.
+
+If at all possible, features **must** be unit tested.
+All tests should have `@covers` tags pointing to the specific code which is being tested.
+Code coverage is monitored for every PR and for the code base as a whole using [CodeCov][].
+
+[PHPUnit]: https://phpunit.readthedocs.io/en/stable/
```

Comment:
The `stable` path unfortunately doesn't seem to work and produces a broken link.
Should we link to 9.5 instead? This would be the version that we are writing the tests against with the polyfills. Updating the polyfills would then mean checking the docs link here if it needs updating...
```suggestion
[PHPUnit]: https://phpunit.readthedocs.io/en/9.5/
```

Reply:
Updated to the new canonical `main`.